python/pyarrow/tests/test_array.py

```diff
@@ -3381,3 +3381,15 @@ def test_struct_array_sort():
         {"a": 5, "b": "foo"},
         None
     ]
+
+
+def test_array_accepts_pyarrow_array():
+    arr = pa.array([1, 2, 3])
+    result = pa.array(arr)
+
+    assert arr == result
+
+    arr_uint = pa.array([1, 2, 3], type=pa.uint8())
+    result = pa.array(arr, type=pa.uint8())
```

Comment:
Can you add an extra test where the passed `type` is different? (eg int64 in this case)

Reply:
You mean adding a test where `arr` type and type passed is different? I do not think I follow - it should be the case here (`arr` is of type int64 and the test passes uint8 type to check the casting?)

I will rename `arr_uint` to `expected` just to make it clearer.

Reply:
Ah, yes, I was misreading the `arr_uint` as the array being passed to `pa.array(..)`